Failure to run swig for M2Crypto on Python 2.6 in Travis

Since today, we have Python 2.6 failures again in Travis.

Symptom:
```
SWIG/_evp.i:12: Error: Unable to find 'openssl/opensslconf.h'
```

A recent successful run: https://travis-ci.org/pywbem/pywbem/jobs/270612220
A failed run: https://travis-ci.org/pywbem/pywbem/jobs/270904030

The major change seems to be that Travis upgraded from Ubuntu 12.04 to 14.04 between these two runs. One change is that openssl-devel was upgraded from 1.0.1 to 1.0.1f.

More details on the symptom:
```
$ make develop
python setup.py develop_os
running develop_os
develop_os: Installing prerequisite OS packages for platform: ubuntu
Processing OS package requirement: libssl-dev>=1.0.1
Installed OS package version is sufficient: libssl-dev 1.0.1f
Processing OS package requirement: g++>=4.4
Installed OS package version is sufficient: g++ 4.8.2
Testing for availability of Swig >=2.0 in PATH...
Swig is available in PATH; testing its version...
Installed Swig version is sufficient: 2.0.11
Processing OS package requirement: python-dev
Installed OS package version is sufficient: python-dev 2.7.5
Processing OS package requirement: libxml2-dev
Installed OS package version is sufficient: libxml2-dev 2.9.1+dfsg1
Processing OS package requirement: libxml2-utils
Running: sudo apt-get install -y  libxml2-utils
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  libxml2-utils
0 upgraded, 1 newly installed, 0 to remove and 101 not upgraded.
Need to get 34.7 kB of archives.
After this operation, 189 kB of additional disk space will be used.
Get:1 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libxml2-utils amd64 2.9.1+dfsg1-3ubuntu4.9 [34.7 kB]
Fetched 34.7 kB in 0s (0 B/s)
Selecting previously unselected package libxml2-utils.
(Reading database ... 93482 files and directories currently installed.)
Preparing to unpack .../libxml2-utils_2.9.1+dfsg1-3ubuntu4.9_amd64.deb ...
Unpacking libxml2-utils (2.9.1+dfsg1-3ubuntu4.9) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up libxml2-utils (2.9.1+dfsg1-3ubuntu4.9) ...
Processing OS package requirement: libxslt1-dev
Installed OS package version is sufficient: libxslt1-dev 1.1.28
Processing OS package requirement: libyaml-dev
Installed OS package version is sufficient: libyaml-dev 0.1.4
Processing OS package requirement: make
Installed OS package version is sufficient: make 3.81
Processing OS package requirement: tar
Installed OS package version is sufficient: tar 1.27.1
Processing OS package requirement: git
Installed OS package version is sufficient: git 2.13.0
python setup.py develop
running develop
running egg_info
creating pywbem.egg-info
writing requirements to pywbem.egg-info/requires.txt
writing pywbem.egg-info/PKG-INFO
writing top-level names to pywbem.egg-info/top_level.txt
writing dependency_links to pywbem.egg-info/dependency_links.txt
writing manifest file 'pywbem.egg-info/SOURCES.txt'
reading manifest file 'pywbem.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pywbem.egg-info/SOURCES.txt'
running build_ext
Creating /home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/pywbem.egg-link (link to .)
Adding pywbem 0.11.0.dev0 to easy-install.pth file
Installing wbemcli script to /home/travis/virtualenv/python2.6.9/bin
Installing wbemcli.py script to /home/travis/virtualenv/python2.6.9/bin
Installing mof_compiler script to /home/travis/virtualenv/python2.6.9/bin
Installing wbemcli.bat script to /home/travis/virtualenv/python2.6.9/bin
Installing mof_compiler.bat script to /home/travis/virtualenv/python2.6.9/bin
Installed /home/travis/build/pywbem/pywbem
Processing dependencies for pywbem==0.11.0.dev0
Searching for M2Crypto>=0.24
Reading https://pypi.python.org/simple/M2Crypto/
Downloading https://pypi.python.org/packages/11/29/0b075f51c38df4649a24ecff9ead1ffc57b164710821048e3d997f1363b9/M2Crypto-0.26.0.tar.gz#md5=9f02f0b88fbe225cc6ea8680945cafa0
Best match: M2Crypto 0.26.0
Processing M2Crypto-0.26.0.tar.gz
Writing /tmp/easy_install-grDliX/M2Crypto-0.26.0/setup.cfg
Running M2Crypto-0.26.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-grDliX/M2Crypto-0.26.0/egg-dist-tmp-BZTgkk
SWIG/_evp.i:12: Error: Unable to find 'openssl/opensslconf.h'
SWIG/_rc4.i:5: Error: Unable to find 'openssl/opensslconf.h'
SWIG/_ec.i:7: Error: Unable to find 'openssl/opensslconf.h'
error: Setup script exited with error: command 'swig' failed with exit status 1
make: *** [develop] Error 1
The command "make develop" failed and exited with 2 during .
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,6 +2,11 @@
 # Must be located in the root directory of the Git repository.
 
 sudo: required
+
+# Disabling the distro, so we get the latest:
+# dist:
+#   - trusty
+
 language: python
 python:
   - "2.6"
@@ -14,14 +19,20 @@ python:
 install:
   - python -c "import platform; print('Demonstration of incorrect distro returned by platform.linux_distribution():'); print(repr(platform.linux_distribution()))"
   - python -c "import pip, os; print('Site-packages directory of system Python:'); print(os.path.dirname(os.path.dirname(pip.__file__)))"
+  - env |sort
   - pip list
 # TODO: Replace the following quick fix for installing swig with a real fix.
-  - sudo apt-get install -y swig
+  - if [[ $TRAVIS_OS_NAME == linux ]]; then sudo apt-get install -y swig; fi
 # end of swig quick fix
   - pip install --upgrade pip
   - python uninstall_pbr_on_py26.py
   - make clobber
   - pip list
+# TODO: Replace the following quick fix for installing M2Crypto with a real fix.
+# The problem happens only with Python 2.6 on Ubuntu 16.04, see pywbem issue #750.
+  - if [[ $TRAVIS_PYTHON_VERSION == 2.6 && -f /usr/include/x86_64-linux-gnu/openssl/opensslconf.h && ! -f /usr/include/openssl/opensslconf.h ]]; then echo "Linking openssl/opensslconf.h"; sudo ln -s /usr/include/x86_64-linux-gnu/openssl/opensslconf.h /usr/include/openssl/opensslconf.h; fi
+  - find /usr/include -name "openssl*.h" -ls 2>/dev/null || true
+# end of m2crypto quick fix
   - make develop
   - pip install python-coveralls
 # Update coverage because the version defined for python-coveralls breaks

